docs/logging.md

```diff
@@ -0,0 +1,134 @@
+# Logging
+
+Thanos supports request logging via flags.
+
+Components are either configured using `--request.logging-config-file` to reference to the configuration file or `--request.logging-config` to provide configuration as YAML directly.
+
+## Configuration
+
+Configuration can be supplied globally, which applies to both `grpc` and `http`. Alternatively `http` and/or `grpc` can be configured independently.
+
+### Options
+
+Valid `level` for `options` should be one of `INFO`, `WARNING`, `ERROR` or `DEBUG`.
+
+```yaml
+options:
+  level: ERROR
+```
+
+The request logging is configurable per phase via the `decision`.
+
+Valid decision combinations are as follows:
+
+* Log details only when the request completes - Log level is determined by the status code from the response.
+
+```yaml
+options:
+  decision:
+    log_start: false
+    log_end: true
+```
+
+* Log details prior to the request and when the request completes - Pre request log is made at `debug` level.
+
+```yaml
+options:
```

Comment:
I think it would be worth adding what the default behaviour is here. From my testing, I think the default behaviour is:

```
options:
  level: DEBUG
  decision:
    log_start: true
    log_end: true
```

As the behaviour seems to be the same with or without this if `--log-level=debug` is specified?

Reply:
IIUC - the default is `ERROR` - https://github.com/thanos-io/thanos/blob/main/pkg/logging/grpc.go#L45